edge-modules/api-proxy-module/rust-sdk/azure-iot-mqtt/src/device.rs

```diff
@@ -0,0 +1,385 @@
+//! This module contains the device client and device message types.
+
+/// A client for the Azure IoT Hub MQTT protocol. This client receives device-level messages.
+///
+/// A `Client` is a [`Stream`] of [`Message`]s. These messages contain twin state messages, direct method requests and cloud-to-device messages.
+///
+/// It automatically reconnects if the connection to the server is broken. Each reconnection will yield one [`Message::TwinInitial`] message.
+pub struct Client {
```

Comment:
q: is this something that you've just created? If it meant to be a generic version of Device SDK for IoT Edge, maybe we should move it outside and make a crate? @dmolokanov your thoughts?

Reply:
Good question, I am not sure what to do with it. You guessed correctly, this is from Arnav but it is in private repo